### DISCUSSION TITLE
Disable TAB to be copilot completion command

### DISCUSSION THREAD

**Discussion by MaxLeb** [AstroNvim/astrocommunity#1274] [Q&A]
I am new to AstroNVim and I am not sure how to disable the behavior of the TAB key for the copilot plugin.

When editing code, it is not practical to have the TAB complete the copilot suggestion while you just want to insert a TAB.

I have tried to create a copilot.lua file in the plugins folder but I don't now how to edit so I can remap the TAB key or at least not override it.

I have seen in the code that it is actually a function defined for nvim-cmp so should I override this one instead of copilot ?

**Comment by MaxLeb** [comment]
Answering to myself since I found a way: 

I just copied https://github.com/AstroNvim/astrocommunity/blob/main/lua/astrocommunity/completion/copilot-lua-cmp/init.lua into my own `lua/plugins/copilot.lua` and edited `["<Tab>"]` to `["C-e"]`

However I still have sometimes (quite often actually) the shotcut to complete copilot suggestion that does not work. 
I haven't find why, I just know that closing and reopening nvim usually makes it work again.

**Comment by luxus** [comment]
sorry .. i guess its quicker to get answers on discord.

you don't have to copy the whole init file. you can just keep the line you want to overwrite

**Comment by luxus** [comment]
even the readme shows a example how to do it https://github.com/AstroNvim/astrocommunity/blob/main/README.md#customizing-astrocommunity-installations

**Reply by MaxLeb** [reply]
Yes but it shows an example for a simple plugin, in the case of `copilot-lua-cmp` the shortcut seems to be done within the `nvim-cmp` dependency and I am not sure how to properly override it.
